src/popup/index.ts

```diff
@@ -1,49 +1,176 @@
+import browser from "webextension-polyfill";
+import { getStreakAndIncrement } from "../background/services/streak";
 import {
   changeSelectedTime,
   handleStartTimer,
+  stopTimer,
 } from "../background/services/timer";
-import browser from 'webextension-polyfill';
-
 import changePopupColor from "../utils/change-popup-color";
 import playSound from "../utils/play-popup-sounds";
 
 const configButton = document.querySelector("#config") as HTMLElement;
 const selectTime = document.querySelector("#select-time") as HTMLSelectElement;
-const timer = document.querySelector("#timer-counter") as HTMLElement;
+const timerDisplay = document.querySelector("#timer-counter") as HTMLElement;
 const startButton = document.querySelector("#start") as HTMLElement;
 const streakCounter = document.querySelector("#streak-counter") as HTMLElement;
+const customInput = document.querySelector("#custom-input") as HTMLInputElement;
+
+let isTimerRunning = false;
 
 function changeAppStyleMode(isRunning: boolean) {
   changePopupColor(isRunning);
   startButton.innerHTML = isRunning ? "GIVE UP!" : "START FOCUSING";
   selectTime.disabled = isRunning;
+  customInput.disabled = isRunning;
 }
 
 function updateTimer() {
-  browser.storage.local.get(
-    ["timer", "selectedTime", "isRunning", "streak"])
+  browser.storage.local
+    .get(["timer", "selectedTime", "timeLabel", "isRunning", "streak"])
     .then((res) => {
-      selectTime.value = res.selectedTime || 25;
-      streakCounter.innerHTML = res.streak || 0;
+      const { timer, selectedTime, timeLabel, isRunning, streak } = res;
 
-      let minutes = `${res.selectedTime - Math.ceil(res.timer / 60)}`.padStart(
-        2,
-        "0"
+      selectTime.value = selectedTime.toString() || "60";
+      streakCounter.innerHTML = streak.toString() || "0";
+
```

Comment:
When the popup is opened again, there is no check if the timer is already running, so the timerIsRunning variable will always be set to false and the user will be able to select the input even with the timer running.

Adding ```timerIsRunning = isRunning``` here should fix it.

Reply:
I made a few more changes, but it looks like I solved the problem. Dealing with states in the application without using a framework can be a bit annoying, but I appreciate the challenge as it encourages us to think about alternative solutions.